binding/flutter/ios/Classes/SwiftPorcupinePlugin.swift

```
@@ -32,32 +31,61 @@ public class SwiftPorcupinePlugin: NSObject, FlutterPlugin {
+                    
+                    var keywordValues: [Porcupine.BuiltInKeyword] = []
+                    for keyword in keywords {
+                        if let builtIn = Porcupine.BuiltInKeyword(rawValue: keyword.capitalized) {
+                            keywordValues.append(builtIn)
+                        } else {
+                            result(errorToFlutterError(PorcupineError.PorcupineKeyError("'\(keyword.lowercased())' is not a built in keyword")))
+                            return
+                        }
+                    }
+                    
+                    let porcupine = try Porcupine(
+                        accessKey: accessKey,
+                        keywords: keywordValues,
+                        modelPath: modelPath,
+                        sensitivities: sensitivities)
                     
-            let keywordPaths = Bundle(for: Porcupine.self).paths(forResourcesOfType: "ppn", inDirectory: nil)
-            var keywordDict:Dictionary<String, String> = [:]
-            for keywordPath in keywordPaths{
-                let keywordName = URL(fileURLWithPath:keywordPath).lastPathComponent.components(separatedBy:"_")[0]
-                keywordDict[keywordName] = keywordPath
+                    let handle: String = String(describing: porcupine)
+                    porcupinePool[handle] = porcupine
+                    
+                    var param: [String: Any] = [:]
+                    param["handle"] = handle
+                    param["frameLength"] = Porcupine.frameLength
+                    param["sampleRate"] = Porcupine.sampleRate
+                    param["version"] = Porcupine.version
+                    
+                    result(param)
+                } else {
+                    result(errorToFlutterError(PorcupineError.PorcupineInvalidArgumentError("Invalid argument provided to Porcupine 'from_built_in_keywords'")))
```

this gets hit if accessKey or keywords are null right? Maybe just be more specific in the throw here and say something like "missing required arguments accessKey and keywords"